test_unstructured/partition/test_html_partition.py

```diff
@@ -723,3 +726,43 @@ def test_partition_html_with_table_without_tbody(tag: str, expected: str):
     )
     partitions = partition_html(text=table_html)
     assert partitions[0].metadata.text_as_html == expected
+
+
+@pytest.fixture
+def partitioned_html_with_duplicate_elements():
+    filename = "example-docs/fake-html-duplicates.html"
+    with open(filename) as f:
+        elements = partition_html(file=f)
+    return elements
+
+
+def test_each_element_has_page_number_and_index_metadata(partitioned_html_with_duplicate_elements):
+    for element in partitioned_html_with_duplicate_elements:
+        assert element.metadata.page_number is not None
+        assert element.metadata.index_on_page is not None
```

Comment:
This test seems a little weak. There's no indication that page_number is an int or index_on_page is not str or something. I think it would be better as something like:
```python
page_index_pairs = [
    (e.metadata.page_number, e.metadata.index_on_page)
    for e in partition_html("example-docs/fake-html-duplicates.html")
]
assert page_index_pairs == [(1, 0), (1, 1), ...]
```

Also this give the user some concrete intuitive sense of how these values proceed along an element-stream.

Reply:
👍 you're right, my implementation was quite naive and needs to be refactored the way you proposed